[BUG] Building warnings after make_strings_column and chars deprecated

**Describe the bug**
There are many warnings when building jni after cudf deprecated `cudf::make_strings_column` with typed offsets and `cudf::strings_column_view::chars`.

```
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__84e76f6e_15_format_float_cu_a43477b9::dispatch_format_float_fn::operator()(const cudf::column_view&, int, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) const [with FloatType = float; std::enable_if_t<is_floating_point_v<FloatType> >* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:486:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__84e76f6e_15_format_float_cu_a43477b9::dispatch_format_float_fn; Ts = {const cudf::column_view&, const int&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu:116:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu:90:33: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]    90 |     return cudf::make_strings_column(strings_count,
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~                                                             
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__84e76f6e_15_format_float_cu_a43477b9::dispatch_format_float_fn::operator()(const cudf::column_view&, int, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) const [with FloatType = double; std::enable_if_t<is_floating_point_v<FloatType> >* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:489:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__84e76f6e_15_format_float_cu_a43477b9::dispatch_format_float_fn; Ts = {const cudf::column_view&, const int&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu:116:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/format_float.cu:90:33: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]    90 |     return cudf::make_strings_column(strings_count,
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~                                                             
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] [15/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/cast_decimal_to_string.cu.o
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn::operator()(const cudf::column_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) const [with T = numeric::fixed_point<int, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:534:79:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn; Ts = {const cudf::column_view&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:215:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:192:27: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   192 |     return make_strings_column(input.size(),
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~                                                                    
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn::operator()(const cudf::column_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) const [with T = numeric::fixed_point<long int, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:537:79:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn; Ts = {const cudf::column_view&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:215:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:192:27: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   192 |     return make_strings_column(input.size(),
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~                                                                    
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn::operator()(const cudf::column_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) const [with T = numeric::fixed_point<__int128, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:540:80:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__ed312537_25_cast_decimal_to_string_cu_d20c84b5::dispatch_decimal_to_non_ansi_string_fn; Ts = {const cudf::column_view&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:215:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_decimal_to_string.cu:192:27: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   192 |     return make_strings_column(input.size(),
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~                                                                    
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] [16/56] Building CXX object CMakeFiles/cudfjnistub.dir/src/emptyfile.cpp.o
[INFO]      [exec] [17/56] Building CXX object CMakeFiles/cufilejni.dir/src/emptyfile.cpp.o
[INFO]      [exec] [18/56] Building CXX object CMakeFiles/spark_rapids_jni.dir/src/GpuTimeZoneDBJni.cpp.o
[INFO]      [exec] [19/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/datetime_rebase.cu.o
[INFO]      [exec] [20/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/utilities.cu.o
[INFO]      [exec] [21/56] Building CXX object tests/CMakeFiles/CAST_DECIMAL_TO_STRING.dir/cast_decimal_to_string.cpp.o
[INFO]      [exec] [22/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/timezones.cu.o
[INFO]      [exec] [23/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/cast_string_to_float.cu.o
[INFO]      [exec] [24/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/zorder.cu.o
[INFO]      [exec] [25/56] Building CXX object tests/CMakeFiles/CAST_STRING.dir/cast_string.cpp.o
[INFO]      [exec] [26/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/xxhash64.cu.o
[INFO]      [exec] [27/56] Building CXX object tests/CMakeFiles/FORMAT_FLOAT.dir/format_float.cpp.o
[INFO]      [exec] [28/56] Building CUDA object faultinj/CMakeFiles/cufaultinj.dir/faultinj.cu.o
[INFO]      [exec] [29/56] Building CXX object tests/CMakeFiles/CAST_FLOAT_TO_STRING.dir/cast_float_to_string.cpp.o
[INFO]      [exec] [30/56] Linking CUDA shared library faultinj/libcufaultinj.so
[INFO]      [exec] [31/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/cast_float_to_string.cu.o
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__e6f99592_23_cast_float_to_string_cu_bf41fb93::dispatch_float_to_string_fn::operator()(const cudf::column_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with FloatType = float; std::enable_if_t<is_floating_point_v<FloatType> >* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:486:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__e6f99592_23_cast_float_to_string_cu_bf41fb93::dispatch_float_to_string_fn; Ts = {const cudf::column_view&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu:113:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu:89:27: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]    89 |     return make_strings_column(strings_count,
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~                                                                   
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::_GLOBAL__N__e6f99592_23_cast_float_to_string_cu_bf41fb93::dispatch_float_to_string_fn::operator()(const cudf::column_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with FloatType = double; std::enable_if_t<is_floating_point_v<FloatType> >* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:489:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::_GLOBAL__N__e6f99592_23_cast_float_to_string_cu_bf41fb93::dispatch_float_to_string_fn; Ts = {const cudf::column_view&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu:113:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_float_to_string.cu:89:27: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]    89 |     return make_strings_column(strings_count,
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~                                                                   
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
[INFO]      [exec]       |                                          ^~~~~~~~~~~~~~~~~~~
[INFO]      [exec] [32/56] Building CXX object tests/CMakeFiles/UTILITIES.dir/utilities.cpp.o
[INFO]      [exec] [33/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/parse_uri.cu.o
[INFO]      [exec] [34/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/cast_string.cu.o
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = signed char; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:462:74:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = short int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:465:75:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:468:75:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = long int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:471:75:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = unsigned char; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:474:75:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = short unsigned int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:477:76:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = unsigned int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:480:76:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = long unsigned int; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:483:76:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = float; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:486:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = double; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:489:77:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_integer_impl::operator()(const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = bool; std::enable_if_t<is_numeric<T>()>* <anonymous> = 0]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:492:75:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_integer_impl; Ts = {const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:794:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:667:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   667 |     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_decimal_impl::operator()(cudf::data_type, int32_t, const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = numeric::fixed_point<int, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0; int32_t = int]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:534:79:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_decimal_impl; Ts = {cudf::data_type&, int&, const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:834:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_decimal_impl::operator()(cudf::data_type, int32_t, const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = numeric::fixed_point<long int, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0; int32_t = int]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:537:79:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_decimal_impl; Ts = {cudf::data_type&, int&, const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:834:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu: In instantiation of 'std::unique_ptr<cudf::column> spark_rapids_jni::detail::string_to_decimal_impl::operator()(cudf::data_type, int32_t, const cudf::strings_column_view&, bool, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [with T = numeric::fixed_point<__int128, numeric::Radix::BASE_10>; std::enable_if_t<is_fixed_point<T>()>* <anonymous> = 0; int32_t = int]':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/utilities/type_dispatcher.hpp:540:80:   required from 'constexpr decltype(auto) cudf::type_dispatcher(cudf::data_type, Functor, Ts&& ...) [with IdTypeMap = cudf::id_to_type_impl; Functor = spark_rapids_jni::detail::string_to_decimal_impl; Ts = {cudf::data_type&, int&, const cudf::strings_column_view&, bool&, bool&, rmm::cuda_stream_view&, rmm::mr::device_memory_resource*&}]'
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:834:23:   required from here
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/cast_string.cu:736:160: warning: 'cudf::column_view cudf::strings_column_view::chars(rmm::cuda_stream_view) const' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   736 |     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
[INFO]      [exec]       |                                                                                                                                                                ^
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/strings/strings_column_view.hpp:113:42: note: declared here
[INFO]      [exec]   113 |   [[deprecated]] [[nodiscard]] column_view chars(
[INFO]      [exec]       |                                          ^~~~~
[INFO]      [exec] [35/56] Building CXX object tests/CMakeFiles/DATETIME_REBASE.dir/datetime_rebase.cpp.o
[INFO]      [exec] [36/56] Building CXX object tests/CMakeFiles/TIMEZONES.dir/timezones.cpp.o
[INFO]      [exec] [37/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/murmur_hash.cu.o
[INFO]      [exec] [38/56] Building CXX object tests/CMakeFiles/PARSE_URI.dir/parse_uri.cpp.o
[INFO]      [exec] [39/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/histogram.cu.o
[INFO]      [exec] [40/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/decimal_utils.cu.o
[INFO]      [exec] [41/56] Building CUDA object tests/CMakeFiles/BLOOM_FILTER.dir/bloom_filter.cu.o
[INFO]      [exec] [42/56] Building CXX object tests/CMakeFiles/HASH.dir/hash.cpp.o
[INFO]      [exec] [43/56] Building CUDA object CMakeFiles/spark_rapids_jni.dir/src/map_utils.cu.o
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/map_utils.cu: In function 'std::unique_ptr<cudf::column> spark_rapids_jni::_GLOBAL__N__7870ef5e_12_map_utils_cu_dc61b5be::extract_keys_or_values(bool, int64_t, const rmm::device_uvector<thrust::pair<unsigned int, unsigned int> >&, const rmm::device_uvector<signed char>&, const rmm::device_uvector<char>&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*)':
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/src/main/cpp/src/map_utils.cu:578:33: warning: 'std::unique_ptr<cudf::column> cudf::make_strings_column(cudf::size_type, std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>, cudf::size_type, rmm::device_buffer&&)' is deprecated [-Wdeprecated-declarations]
[INFO]      [exec]   578 |   return cudf::make_strings_column(
[INFO]      [exec]       |        ~~~~~~~~~~~~~~~~~~~~~~~~~^~~                                                                             
[INFO]      [exec] /home/haoyangl/spark-rapids-jni/target/libcudf-install/include/cudf/column/column_factories.hpp:465:42: note: declared here
[INFO]      [exec]   465 | [[deprecated]] std::unique_ptr<column> make_strings_column(size_type num_strings,
```

**Expected behavior**
No warnings when building.

--- a/src/main/cpp/benchmarks/common/generate_input.cu
+++ b/src/main/cpp/benchmarks/common/generate_input.cu
@@ -520,7 +520,7 @@ std::unique_ptr<cudf::column> create_random_utf8_string_column(data_profile cons
   return cudf::make_strings_column(
     num_rows,
     std::move(offsets),
-    std::move(chars),
+    std::move(chars->release().data.release()[0]),
     profile.get_null_frequency().has_value() ? std::move(result_bitmask) : rmm::device_buffer{},
     null_count);
 }

--- a/src/main/cpp/src/cast_decimal_to_string.cu
+++ b/src/main/cpp/src/cast_decimal_to_string.cu
@@ -191,7 +191,7 @@ struct dispatch_decimal_to_non_ansi_string_fn {
 
     return make_strings_column(input.size(),
                                std::move(offsets),
-                               std::move(chars),
+                               std::move(chars->release().data.release()[0]),
                                input.null_count(),
                                cudf::detail::copy_bitmask(input, stream, mr));
   }

--- a/src/main/cpp/src/cast_float_to_string.cu
+++ b/src/main/cpp/src/cast_float_to_string.cu
@@ -88,7 +88,7 @@ struct dispatch_float_to_string_fn {
 
     return make_strings_column(strings_count,
                                std::move(offsets),
-                               std::move(chars),
+                               std::move(chars->release().data.release()[0]),
                                floats.null_count(),
                                cudf::detail::copy_bitmask(floats, stream, mr));
   }

--- a/src/main/cpp/src/cast_string.cu
+++ b/src/main/cpp/src/cast_string.cu
@@ -624,7 +624,7 @@ void validate_ansi_column(column_view const& col,
     dest.resize(string_bounds[1] - string_bounds[0]);
 
     cudaMemcpyAsync(dest.data(),
-                    &source_col.chars().data<char const>()[string_bounds[0]],
+                    &source_col.chars_begin(stream)[string_bounds[0]],
                     string_bounds[1] - string_bounds[0],
                     cudaMemcpyDeviceToHost,
                     stream.value());
@@ -667,7 +667,7 @@ struct string_to_integer_impl {
     detail::string_to_integer_kernel<<<blocks, threads, 0, stream.value()>>>(
       data.data(),
       null_mask.data(),
-      string_col.chars().data<char const>(),
+      string_col.chars_begin(stream),
       string_col.offsets().data<size_type>(),
       string_col.null_mask(),
       string_col.size(),
@@ -736,7 +736,7 @@ struct string_to_decimal_impl {
     detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
       data.data(),
       null_mask.data(),
-      string_col.chars().data<char const>(),
+      string_col.chars_begin(stream),
       string_col.offsets().data<size_type>(),
       string_col.null_mask(),
       string_col.size(),

--- a/src/main/cpp/src/format_float.cu
+++ b/src/main/cpp/src/format_float.cu
@@ -89,7 +89,7 @@ struct dispatch_format_float_fn {
 
     return cudf::make_strings_column(strings_count,
                                      std::move(offsets),
-                                     std::move(chars),
+                                     std::move(chars->release().data.release()[0]),
                                      floats.null_count(),
                                      cudf::detail::copy_bitmask(floats, stream, mr));
   }

--- a/src/main/cpp/src/map_utils.cu
+++ b/src/main/cpp/src/map_utils.cu
@@ -575,8 +575,11 @@ std::unique_ptr<cudf::column> extract_keys_or_values(
 
   auto children = cudf::strings::detail::make_strings_children(
     substring_fn{unified_json_buff, extract_ranges}, num_extract, stream, mr);
-  return cudf::make_strings_column(
-    num_extract, std::move(children.first), std::move(children.second), 0, rmm::device_buffer{});
+  return cudf::make_strings_column(num_extract,
+                                   std::move(children.first),
+                                   std::move(children.second->release().data.release()[0]),
+                                   0,
+                                   rmm::device_buffer{});
 }
 
 // Compute the offsets for the final lists of Struct<String,String>.

